python/cudf/cudf/core/dtypes.py

```diff
@@ -488,87 +495,37 @@ class Decimal64Dtype(_BaseDtype):
-
-    @property
-    def precision(self):
-        return self._typ.precision
-
-    @precision.setter
-    def precision(self, value):
-        self._validate(value, self.scale)
-        self._typ = pa.decimal128(precision=value, scale=self.scale)
 
-    @property
-    def scale(self):
-        return self._typ.scale
-
-    @property
-    def type(self):
-        # might need to account for precision and scale here
-        return decimal.Decimal
-
-    def to_arrow(self):
-        return self._typ
-
-    @classmethod
-    def from_arrow(cls, typ):
-        return cls(typ.precision, typ.scale)
-
-    @property
-    def itemsize(self):
-        return 8
-
-    def __repr__(self):
-        return (
-            f"{self.__class__.__name__}"
-            f"(precision={self.precision}, scale={self.scale})"
-        )
-
-    def __hash__(self):
-        return hash(self._typ)
-
-    @classmethod
-    def _validate(cls, precision, scale=0):
-        if precision > Decimal64Dtype.MAX_PRECISION:
-            raise ValueError(
-                f"Cannot construct a {cls.__name__}"
-                f" with precision > {cls.MAX_PRECISION}"
-            )
-        if abs(scale) > precision:
-            raise ValueError(f"scale={scale} exceeds precision={precision}")
-
-    @classmethod
-    def _from_decimal(cls, decimal):
-        """
-        Create a cudf.Decimal64Dtype from a decimal.Decimal object
-        """
-        metadata = decimal.as_tuple()
-        precision = max(len(metadata.digits), -metadata.exponent)
-        return cls(precision, -metadata.exponent)
+class Decimal128Dtype(DecimalDtype):
+    """
+    Parameters
+    ----------
+    precision : int
+        The total number of digits in each value of this dtype
+    scale : int, optional
+        The scale of the Decimal128Dtype. See Notes below.
 
-    def serialize(self) -> Tuple[dict, list]:
-        return (
-            {
-                "type-serialized": pickle.dumps(type(self)),
-                "precision": self.precision,
-                "scale": self.scale,
-            },
-            [],
-        )
+    Notes
+    -----
+        When the scale is positive:
+            - numbers with fractional parts (e.g., 0.0042) can be represented
+            - the scale is the total number of digits to the right of the
+            decimal point
+        When the scale is negative:
+            - only multiples of powers of 10 (including 10**0) can be
+            represented (e.g., 1729, 4200, 1000000)
+            - the scale represents the number of trailing zeros in the value.
+        For example, 42 is representable with precision=2 and scale=0.
+        13.0051 is representable with precision=6 and scale=4,
+        and *not* representable with precision<6 or scale<4.
+    """
 
-    @classmethod
-    def deserialize(cls, header: dict, frames: list):
-        return cls(header["precision"], header["scale"])
+    name = "decimal128"
+    MAX_PRECISION = 38
+    itemsize = 16
```

Comment:
I don't know if we have a convention for class variables being upper or lower case, but I would suggest being consistent at least within these classes in lieu of a broader rule. It is odd to see MAX_PRECISION capitalized while itemsize is lowercase.

Reply:
In `column.pyx` we use `itemsize` attribute to calculate a columns `base_size`:
```python
@property
    def base_size(self):
        return int(self.base_data.size / self.dtype.itemsize)
```

Should we capitalize class constant to `ITEMSIZE` and introduce a property in parent `DecimalType` class that just returns the class constant?

```python
@property
def itemsize(self):
      return self.ITEMSIZE
```

Reply:
I did the above changes, if you think it was unnecessary let me know I'll revert it.

Reply:
I take that back, a `property` does effectively make this a read-only attribute..

Reply:
Ah I missed that `numpy.dtype` has an itemsize attribute and that we needed to match that. @shwina IMO in general we shouldn't use properties over class attributes just to enforce immutability. If something really belongs to the class, not to instances, I think it is more Pythonic to put it in the class and allow users to shoot themselves in the foot by modifying it (as long as we document appropriately). That said, in this case we should match whatever numpy is doing, so @galipremsagar's solution looks fine to me.